src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -87,6 +98,27 @@ public void setMetadataConfig(SnowflakeMetadataConfig metadataConfigIn) {
     metadataConfig = metadataConfigIn;
   }
 
+  /**
+   * extract enableSchematization from the connector config and set the value for the recordService
+   *
+   * <p>The extracted boolean is returned for external usage.
+   *
+   * @param connectorConfig the connector config map
+   * @return a boolean indicating whether schematization is enabled
+   */
+  public boolean setEnableSchematizationFromConfig(final Map<String, String> connectorConfig) {
```

Comment:
can you name this setAndGetEnable
```suggestion
  public boolean setAndGetEnableSchematizationFromConfig(final Map<String, String> connectorConfig) {
```